astroid/brain/brain_typing.py

```diff
@@ -78,6 +78,57 @@ def looks_like_typing_namedtuple(node):
     return False
 
 
+def looks_like_typing_typevar_or_newtype(node):
+    func = node.func
+    if isinstance(func, nodes.Attribute):
+        return func.attrname in {'TypeVar', 'NewType'}
+    if isinstance(func, nodes.Name):
+        return func.name in {'TypeVar', 'NewType'}
+    return False
+
+
+TYPING_TYPE_TEMPLATE = """
+class Meta:
+    def __getitem__(self, item):
```

Comment:
A TypeVar doesn't seem to support indexing. Why do we implement ``__getitem__`` for it?